--- ../src/swipl-devel/packages/clib/time.c	2025-12-17 16:15:00.016552505 +0100
+++ ../src/swipl-devel/packages/clib/time.c	2025-12-19 18:11:19.841606195 +0100
@@ -403,11 +403,12 @@
 }
 
 
-static void
-cleanup_thread(void *data)
+static int
+cleanup_thread(void *unused1, unsigned int unused2)
 { schedule *sched = TheSchedule();
 
-  (void)data;
+  (void)unused1;
+  (void)unused2;
 
   if ( sched->first )
   { Event ev, next;
@@ -428,6 +429,7 @@
     pthread_cond_signal(&cond);
     UNLOCK();
   }
+  return true;
 }
 
 
