--- ../src/swipl-devel/src/pl-dde.c	2025-12-19 19:25:50.166707964 +0100
+++ ../src/swipl-devel/src/pl-dde.c	2025-12-11 21:12:24.228435454 +0100
@@ -363,8 +363,8 @@
 }
 
 #ifdef O_PLMT
+static int
+dde_uninitialise(void *unused1, unsigned long unused2)
-static void
-dde_uninitialise(void *closure)
 { GET_LD
   DWORD ddeInst;
 
@@ -379,6 +379,7 @@
   FUNCTOR_dde_disconnect1 = 0;
   FUNCTOR_dde_request4 = 0;
   FUNCTOR_dde_execute3 = 0;
+  return 0;
 }
 #endif
 
