--- ../src/swipl-devel/tests/compile/test_load_context.pl	2026-09-22 21:08:27.375572560 +0200
+++ ../src/swipl-devel/tests/compile/test_load_context.pl	2026-09-23 22:46:52.710300022 +0200
@@ -64,11 +64,11 @@
        cleanup(remove_files([Parent, Part]))
      ]) :-
     ensure_loaded(user:Parent),
-    once(source_file_property(Part, load_context(Module, Location, _Options))),
+    once(source_file_property(Part, load_context(Module, Location:_, _Options))),
     unload_file(Parent),
     unload_file(Part),
     assertion(Module == test_load_context_parent),
-    assertion(Location = Parent:_).
+    assertion(same_file(Location, Parent)).
 
 :- end_tests(load_context).
 
