--- ../src/swipl-devel/tests/save/test_saved_states.pl	2025-11-10 23:24:20.742422715 +0100
+++ ../src/swipl-devel/tests/save/test_saved_states.pl	2025-11-10 23:24:05.760857798 +0100
 %!  state_output(+Id, -FileName)
 %
 %   Name of the file we use for temporary output of the state.
@@ -120,7 +142,7 @@
     format(atom(File), 'test_state_~w_~w.exe', [Id, Pid]),
     directory_file_path(Dir, File, State).
 
-me(Exe) :-
+me(Exe, Args) :-
     current_prolog_flag(executable, WinExeOS),
     prolog_to_os_filename(WinExe, WinExeOS),
     file_base_name(WinExe, WinFile),
@@ -128,14 +150,28 @@
     !,
     file_directory_name(WinExe, WinDir),
     atomic_list_concat([WinDir, 'swipl.exe'], /, PlExe),
-    prolog_to_os_filename(PlExe, Exe).
-me(MeSH) :-
-    absolute_file_name('swipl.sh', MeSH,
+    prolog_to_os_filename(PlExe, Exe),
+    Args = [].
+me(Cmd, Args) :-
+    current_prolog_flag(windows, true),
+    absolute_file_name('swipl.bat', MeSH0, 
+                       [ access(execute),
+                         file_errors(fail) 
+                       ]),
+    !,
+    Cmd = path('cmd.exe'),
+    prolog_to_os_filename(MeSH0, MeSH),
+    Args = ['/c', MeSH].
+me(MeSH, Args) :-
+    \+ current_prolog_flag(windows, true),
+    absolute_file_name('swipl.sh', MeSH0,
                        [ access(execute),
                          file_errors(fail)
                        ]),
-    !.
-me(Exe) :-
+    !,
+    MeSH = MeSH0,
+    Args = [].
+me(Exe, []) :-
     current_prolog_flag(executable, Exe).
 
 :- dynamic
@@ -156,8 +191,8 @@
 set_windows_path.
 
 create_state(File, Output, Args) :-
-    me(Me),
-    append(Args, ['-o', Output, '-c', File, '-f', none], AllArgs),
+    me(Me, Args0),
+    append([Args0, Args, ['-o', Output, '-c', File, '-f', none]], AllArgs),
     test_dir(TestDir),
     debug(save, 'Creating state in ~q using ~q ~q', [TestDir, Me, AllArgs]),
     process_create(Me, AllArgs,
@@ -173,7 +208,10 @@
     debug(save, 'Running state ~q ~q', [Exe, Args]),
     set_windows_path,
     current_prolog_flag(home, HOME), % needed for MSYS2
-    process_create(Exe, Args,
+    me(Me, Args0),
+    append([Args0, ['-x', Exe, '--'], Args], AllArgs),
+    atomic_list_concat(AllArgs, ' ', ArgStr),
+    process_create(Me, [ArgStr],
                    [ stdout(pipe(Out)),
                      stderr(pipe(Err)),
                      environment(['SWI_HOME_DIR'=HOME])
