--- ../src/swipl-devel/src/pl-cont.c	2025-12-17 16:11:07.791145719 +0100
+++ ../src/swipl-devel/src/pl-cont.c	2025-12-19 21:36:44.522868399 +0100
@@ -486,8 +486,7 @@
     cl = cref->value.clause;
     pcoffset = (size_t)valInt(*ep++);
 
-    lneeded = sizeof(word) +
-	      (size_t)argFrameP((LocalFrame)NULL, cl->variables);
+    lneeded = sizeof(word) + (size_t)argFrameP0(cl->variables);
     lroom   = roomStack(local);
     if ( unlikely(lroom < lneeded) )	/* resize the stack */
     { int rc;
