--- ../src/swipl-devel/src/pl-incl.h	2025-12-17 16:11:07.795121520 +0100
+++ ../src/swipl-devel/src/pl-incl.h	2025-12-19 21:37:23.384484601 +0100
@@ -1159,6 +1159,7 @@
 #define setLevelFrame(fr, l)	do { (fr)->level = (l); } while(0)
 #define levelFrame(fr)		((fr)->level)
 #define argFrameP(f, n)		((Word)((f)+1) + (n))
+#define argFrameP0(n)		((Word)(ARGOFFSET) + (n))
 #define argFrame(f, n)		(*argFrameP((f), (n)) )
 #define varFrameP(f, n)		((Word)(f) + (n))
 #define varFrame(f, n)		(*varFrameP((f), (n)) )
