--- ../src/swipl-devel/tests/save/test_qlf.pl	2026-09-22 21:08:27.394574984 +0200
+++ ../src/swipl-devel/tests/save/test_qlf.pl.new	2026-09-24 08:11:20.241599176 +0200
@@ -156,11 +156,12 @@
      [ setup(compiled_file(Pl, Qlf)),
        cleanup(remove_files([Pl, Qlf]))
      ]) :-
+    '$file_hash'(Pl, Hash),
     '$qlf_sources'(Qlf, Sources),
-    memberchk(source(Pl, Hash), Sources),
+    memberchk(source(Pl1, Hash), Sources),
     Hash =\= 0,
-    '$file_hash'(Pl, Hash).
-
+    same_file(Pl, Pl1).
+    
 %       Reinstalling a tree, or checking it out again, gives every source
 %       a time of its own.  Nothing changed, so nothing is recompiled.
 
@@ -169,6 +170,7 @@
        cleanup(remove_files([Pl, Qlf]))
      ]) :-
     touch(Pl, Qlf, 10),
+    assertion(Pl = Qlf),
     \+ '$qlf_out_of_date'(Pl, Qlf, _).
 
 test(and_one_whose_content_changed_is,
@@ -220,10 +222,11 @@
      [ setup(compiled_with_dependency(Pl, Qlf, Dep)),
        cleanup(remove_files([Pl, Qlf, Dep]))
      ]) :-
+    '$file_hash'(Dep, Hash),
     '$qlf_sources'(Qlf, Sources),
-    memberchk(dependency(Dep, Hash), Sources),
+    memberchk(dependency(Dep1, Hash), Sources),
     Hash =\= 0,
-    '$file_hash'(Dep, Hash).
+    same_file(Dep, Dep1).
 
 test(and_a_change_to_it_needs_a_rebuild,
      [ setup(compiled_with_dependency(Pl, Qlf, Dep)),
