--- ../src/swipl-devel/boot/init.pl	2026-09-24 08:51:28.564842694 +0200
+++ ../src/swipl-devel/boot/init.pl	2026-09-24 08:55:13.119706219 +0200
@@ -2447,9 +2447,11 @@
 
 '$qlf_source_changed'(QlfFile, PlFile) :-
     (   catch('$qlf_sources'(QlfFile, Sources), _, fail),
-	'$member'(source(PlFile, Hash), Sources),
+	'$file_hash'(PlFile, Hash),
 	Hash =\= 0
-    ->  \+ '$file_hash'(PlFile, Hash)
+    ->	\+ ( '$member'(source(PlFile1, Hash), Sources),
+	     same_file(PlFile, PlFile1)
+	   )
     ;   true
     ).
 
